tools/java/runfiles/Runfiles.java

```diff
@@ -376,7 +371,17 @@ public Map<String, String> getEnvVars() {
     return preloadedRunfiles.getEnvVars();
   }
 
-  /** Returns true if the platform supports runfiles only via manifests. */
+  String getCanonicalRepository(String apparentRepositoryName) {
```

Comment:
Can this be made private?

Reply:
It needs to be accessible to Stardoc, which is why I have made it package-private coupled with a target that exposes it publicly but is only visible to Stardoc. I don't see a more restricted way short of `setAccessible` and reflection.